src/main/java/io/leonis/algieba/spatial/BoundaryPotentialField.java

```diff
@@ -0,0 +1,174 @@
+package io.leonis.algieba.spatial;
+
+import io.leonis.algieba.geometry.CardinalDirection;
+import java.util.function.UnaryOperator;
+import lombok.Value;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.ops.transforms.Transforms;
+
+/**
+ * The Class BoundaryPotentialField.
+ *
+ * This class represents a {@link PotentialField} which bounds a rectangle such that the potential
+ * approximates zero for any point on the field and approximates infinity for any
+ * point outside the field.
+ *
+ * @author Rimon Oz
+ */
+@Value
+public class BoundaryPotentialField implements PotentialField {
+  private final INDArray origin = Nd4j.zeros(2, 1);
+  /**
+   * The width of the field in mm.
+   */
+  private final double width;
+  /**
+   * The length of the field in mm.
+   */
+  private final double length;
+  /**
+   * The displacement of the potential field from the edge.
+   */
+  private final double fieldDisplacement;
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public double getPotential(final INDArray positionVector) {
+    return this.getPotential(positionVector.getDouble(0, 0))
+        + this.getPotential(this.getWidth() - positionVector.getDouble(0, 0))
+        + this.getPotential(positionVector.getDouble(1, 0))
+        + this.getPotential(this.getLength() - positionVector.getDouble(1, 0));
+  }
+
+  /**
+   * See <a href="http://www.wolframalpha.com/input/?i=1%2F(1+%2B+e%5E(D+-+d))">this equation.</a>
+   *
+   * @param distanceToBoundary The distance to the boundary in mm.
+   * @return The potential due to a single boundary at the specified distance from that boundary.
+   */
+  public double getPotential(final double distanceToBoundary) {
+    return 1 / (1 + Math.exp(this.getFieldDisplacement() - distanceToBoundary));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public INDArray getForce(final INDArray positionVector) {
+    return this.getForce(positionVector, CardinalDirection.NORTH)
+        .add(this.getForce(positionVector, CardinalDirection.SOUTH))
+        .add(this.getForce(positionVector, CardinalDirection.EAST))
+        .add(this.getForce(positionVector, CardinalDirection.WEST));
+  }
+
+  /**
+   * @param positionVector The position at which to compute the gradient of the potential.
+   * @param direction      The {@link CardinalDirection} matching the boundary for which to compute
+   *                       the gradient of the potential.
+   * @return The gradient of the potential at the specified point due to the boundary located in the
+   *     specified {@link CardinalDirection}.
+   */
+  public INDArray getForce(final INDArray positionVector, final CardinalDirection direction) {
+    switch (direction) {
```

Comment:
Why use a switch case instead of dedicated functions in the original `getForce` method?